docs: Add parsing JSON mixed-up with plain text log message (#8561)
newrelic/docs-website: src/content/docs/logs/ui-data/parsing.mdx

              ISO 8601 time as a `long`
            </td>
          </tr>

          <tr>
            <td>
              `json`
            </td>
            <td>
              JSON structured data, take a look to [Parsing JSON mixed-up with plain text](#parsing-json) for more information
            </td>
          </tr>
        </tbody>
      </table>
  </Collapser>
  <Collapser
    id="parsing-json"
    title="Parsing JSON mixed-up with plain text"
  >
    New Relic logs pipeline parses your log JSON messages by default but sometimes you've JSON log messages that are mixed up with plain text and in that case you also want to be able to parse them and then be able to filter using the JSON attributes.

>     If thats the case you can use the `json` [grok type](#grok-syntax) that will do exactly that, parse the JSON extracted by the grok pattern.

Reviewer: ```suggestion
    If that is the case, you can use the `json` [grok type](#grok-syntax), which will parse the JSON captured by the grok pattern.
```

>     As an example, imagine your logs are JSON logs but prefixed with a date time string:

Reviewer: ```suggestion
    Using the `json` [grok type](#grok-syntax), you can extract and parse JSON from logs that are not properly formatted, e.g, your logs are prefixed with a date time string:
```

    ```
    2015-05-13T23:39:43.945958Z {"event": "TestRequest", "status": 200, "response": {"headers": {"X-Custom": "foo"}, "request": {"headers": {"X-Custom": "bar"}}}}
    ```

>     For parsing that we can create a Parsing Rule with the following Grok:

Reviewer: ```suggestion
    In order to extract and parse the JSON data from this log format, you would create the following Grok expression:
```

    ```
    %{TIMESTAMP_ISO8601:containerTimestamp} %{GREEDYDATA:log:json}
    ```

>     And the log will result as:

Reviewer: ```suggestion
    The resulting log would be:
```

    ```
    containerTimestamp: "2015-05-13T23:39:43.945958Z"
    log.event: "TestRequest"
    log.status: 200
    log.response.headers.X-Custom: "foo"
    log.request.headers.X-Custom: "bar"
    ```

>     It is possible to configure some JSON parser behaviors using `:json(_CONFIG_)`:

Reviewer: ```suggestion
    Optionally, you can configure the `json` [grok type](#grok-syntax) using `:json(_CONFIG_)`:
```

>     - `json({"dropOriginal": true})`: Drop original field, this defaults to true and means that will drop the JSON string attribute after parsing it.

Reviewer: ```suggestion
    - `json({"dropOriginal": true})`: Drop the original attribute. When set to `true` (default value), the Parsing rule will drop the original JSON attribute.
```

>     - `json({"depth": 62})`: Allows to control how much levels you want to parse from your JSON (It defaults to its max that is 62).

Reviewer: ```suggestion
    - `json({"depth": 62})`: Levels of depth you want to parse the JSON value (defaulted to 62).
```
